commerce_coordinator/apps/commercetools/data.py

```diff
@@ -111,7 +111,7 @@ def order_from_commercetools(order: CTOrder, customer: CTCustomer) -> LegacyOrde
         date_placed=order.last_modified_at,
         total_excl_tax=typed_money_to_string(order.total_price, money_as_decimal_string=SEND_MONEY_AS_DECIMAL_STRING),
         # in dev systems, this isn't set... so let's use UUID, otherwise, let's rely on order number
-        number=order.id,  # Long-term: this_or(order.order_number, order.id),
+        number=order.order_number or order.id,
```

Comment:
Since we will be passing and displaying order_number for the order history page, we need to make a change in CC to rely on the order number for the Order receipt request.

Need to change the method to [get_order_by_number](https://github.com/edx/commerce-coordinator/blob/ca054ba93b3d612ea740b5b4868ae96adafbc9c7/commerce_coordinator/apps/commercetools/clients.py#L195)(): https://github.com/edx/commerce-coordinator/blob/ca054ba93b3d612ea740b5b4868ae96adafbc9c7/commerce_coordinator/apps/commercetools/pipeline.py#L85